[BE] Kafka 및 ElasticSearch 연결

KHJHK opened the pull request:
Kafka 및 ElasticSearch 연결 작업 진행

- [x] Kafka 및 ElastiSearch Docker 설정(docker-compose 파일 작성 및 makefile 작성)
- [x] Kafka 및 ElastiSearch 환경설정
- [x] TestCode 작성

실제 로직 구현시 Producer, Consumer Config 설정 추가 및 Listener 관리 비즈니스 로직 필요

KHJHK committed:
#25 - feat : kafka 및 elasticsearch 연결 작업

- kafka 및 elasticsearch 컨테이너 생성 구문 docker-compse.yml에 추가
- application-core.yml에 kafka 및 es 설정 추가
- build.gradle에 kafka, es 추가

KHJHK committed:
Merge branch 'develop' of https://github.com/SIMPAYLOG/finance-data-generator into feat/25-kafka-es-setup

KHJHK committed:
#25 - chore : kafka 및 elasticsearch docker-compose.yml 파일 분리 및 네트워크 그룹화

- kafka 및 elasticsearch docker-compose 파일 분리
- kafka 및 elasticsearch를 kafka-net 네트워크로 그룹화
- kafka 및 elasticsearch 컨테이너를 관리하는 Makefile 생성

KHJHK committed:
#25 - Test : Kafka-Elasticsearch 연결 TestCode 작성

Kafka와 ElasticSearch를 연결하는 TestCode 작성

lmw7414 commented on data/Makefile:
Make 파일은 최상위 루트 디렉토리에서만 사용하는 걸로 해요!

lmw7414 commented on data/elasticsearch/docker-compose.yml:
kafka와 es 네트워크 분리해주세요.

lmw7414 commented on generator-core/src/main/resources/application-core.yml:
core에 필요하지 않은 설정정보입니다. 책임을 분리해주세요.

lmw7414 commented on generator-simulator/src/test/java/com/simpaylog/generatorsimulator/kafka/KafkaToElasticTest.java:
```diff
@@ -0,0 +1,139 @@
+import java.util.Map;
+import java.util.Properties;
+
+@TestPropertySource(properties = {
```
application-core의 test 페이지로 만들어놨는데 왜 추가 했을까요?

lmw7414 commented on generator-simulator/src/test/java/com/simpaylog/generatorsimulator/kafka/KafkaToElasticTest.java:
```diff
@@ -0,0 +1,139 @@
+import java.util.Map;
+import java.util.Properties;
+
+@TestPropertySource(properties = {
+        "spring.datasource.url=jdbc:h2:mem:testdb",  // H2 임시 메모리 DB
+        "spring.datasource.driver-class-name=org.h2.Driver",
+        "spring.datasource.username=sa",
+        "spring.datasource.password=",
+        "spring.jpa.hibernate.ddl-auto=none",  // Hibernate가 테이블 생성 안함
+        "spring.sql.init.mode=never",          // SQL 스크립트(schema.sql 등) 실행 안함
+        "spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.H2Dialect"
+})
```
application-core의 test 페이지로 만들어놨는데 왜 추가 했을까요?

KHJHK replied:
분명 위 코드 없으면 에러났었는데... 왜 지금은 안날까요

KHJHK committed:
#25 - chore : Makefile 변경 elasticsearch 네트워크 분리

- makefile을 최상단 makefile로 통합 및 하위 makefile 삭제
- kafka와 elasticsearch 네트워크 분리
- tsetcode 불필요한 TestPropertySource 제거

KHJHK committed:
#25 - test : KafkaFlowTest 코드 재작성

- Kafka 연결 및 동작을 테스트하는 KafkaFlowTest 작성
- Kafka 테스트를 위한 kafka-test 의존성 추가

lmw7414 commented:
수고 많으셨습니다!
혹시 리팩토링 요소가 있다거나, 추후 개선시켜야 할 방향이 있다면 PR에 추가 내용을 넣어주시면 좋을 것 같아요!

b9s2w6 reviewed (commented):
테스트 코드까지 작성해주셔서 동작 방식이 더 잘 이해되는 것 같습니다!!
